feat(ui): upgrade RMWC

endymion1818 opened the pull request:
## Changes
- Upgraded RMWC components to v7
- Upgrade @material components to versions used in v7 of RMWC
- Import SASS to use new syntax
- Update styles to resolve changes from the upgrade

## How can this be tested?
- https://d3mmrkoy170qpm.cloudfront.net/

endymion1818 committed:
feat(ui): upgrade rmwc/base, use npm package

endymion1818 committed:
feat(ui): upgrade rmwc/base, use npm package

endymion1818 committed:
chore(ui): update components

endymion1818 committed:
chore(ui): update components

endymion1818 committed:
fix(app-page-builder, app-form-builder): update other packages

endymion1818 committed:
fix(app-page-builder, app-form-builder): update other packages

endymion1818 committed:
fix(app-admin): resolve tslib to fixed version due to version clashes

endymion1818 committed:
fix(app-admin): resolve tslib to fixed version due to version clashes

endymion1818 committed:
fix(api): rebase & fix conflicts

endymion1818 committed:
fix(api): rebase & fix conflicts

endymion1818 committed:
fix(settings): add back typescript to resolutions

endymion1818 committed:
fix(settings): add back typescript to resolutions

endymion1818 committed:
chore(ui): update material components

endymion1818 committed:
chore(ui): update material components

endymion1818 committed:
fix(project-utils): turn off webpackImporter

endymion1818 committed:
fix(project-utils): turn off webpackImporter

endymion1818 committed:
fix(material): resolve material versions

endymion1818 committed:
fix(material): resolve material versions

endymion1818 committed:
fix(ui): upgrade material-components-web

endymion1818 committed:
fix(ui): upgrade material-components-web

endymion1818 committed:
fix(app-admin): fix grid layout issue

endymion1818 committed:
fix(app-admin): correct font on body tag

endymion1818 committed:
fix(app-admin): restore installer grid height

endymion1818 committed:
fix(ui): resolvem aterial select to latest vesion

endymion1818 committed:
fix(ui): resolvem aterial select to latest vesion

endymion1818 committed:
fix(ui): update select styles

endymion1818 committed:
fix(ui): update select styles

Pavel910 commented on apps/theme/pageBuilder/styles/medium.scss:
```diff
@@ -1,8 +1,8 @@
  */
-@import "~medium-editor/dist/css/medium-editor.css";
-@import "~medium-editor/dist/css/themes/mani.css";
+@import "medium-editor/dist/css/medium-editor.css";
```
Existing user projects will still have imports via `~`. Will this upgrade break this for those projects, or is this backwards compatible? This has to be tested with older projects.

endymion1818 replied:
It's being deprecated but I couldn't find any notification as to when they might remove it. https://webpack.js.org/loaders/sass-loader/#resolving-import-at-rules

Pavel910 commented on apps/theme/pageBuilder/styles/medium.scss:
```diff
@@ -1,8 +1,8 @@
-@import "~medium-editor/dist/css/medium-editor.css";
-@import "~medium-editor/dist/css/themes/mani.css";
+@import "medium-editor/dist/css/medium-editor.css";
+@import "medium-editor/dist/css/themes/mani.css";
```
`mani.css` ? I have a feeling this should be `main.css`. Please double-check.

endymion1818 replied:
`mani` is in fact the name of the Medium editor theme we're using: https://github.com/yabwe/medium-editor/blob/master/src/sass/themes/mani.scss 😆

Pavel910 commented on apps/theme/styles.scss:
```diff
@@ -1,3 +1,3 @@
 @use "./pageBuilder/styles/theme.scss";
 @use "./formBuilder/styles/theme.scss" as themeFormBuilder;
-@use "~@webiny/react-rich-text-renderer/styles.scss";
+@import "@webiny/react-rich-text-renderer/styles.scss";
```
What's the difference between `@use` and `@import`, and why is `import` only used  on the last scss file, but not the first two as well?

endymion1818 replied:
~We bumped up `sass` and `sass-loader` as part of this upgrade:~

- `@import` globally imports other sass files and [is being phased out](https://sass-lang.com/documentation/at-rules/import). It's appropriate here because we're importing `rich-text-renderer` styles from another package, whereas we're using the other two that come from this package. Also you get errors any other way around 🙃
- `@use` forwards mixins, variables etc _to the current sass file_, so it's more explicit as to what's happening
- `~` (tilde) is deprecated from `sass-loader` (https://webpack.js.org/loaders/sass-loader/#resolving-import-at-rules)

Pavel910 commented on package.json:
```diff
@@ -102,6 +102,7 @@
     "rxjs": "^6.5.5",
     "semver": "^6.3.0",
     "ts-jest": "^26.4.4",
+    "tslib": "2.4.0",
```
Why do we need this `tslib` package? We didn't need it thus far, I want to know what dependency made this a requirement, and if it has to sit in the root package.json.

endymion1818 replied:
it's an implicit dependency of a few projects, and as a result of different versions being used you get this error on building:

`admin: Error: Attempted import error: '__spreadArray' is not exported from 'tslib' (imported as '__spreadArray').`

Unfortunately you still get this error if it's only listed in the `resolutions`.

Pavel910 commented on package.json:
```diff
@@ -211,20 +212,41 @@
     "@types/eslint": "8.2.1",
-    "yargs": "^17.3.1"
+    "yargs": "^17.3.1",
+    "@material/base": "^5.1.0",
+    "@material/dom": "^5.1.0",
+    "@material/textfield": "^5.1.0",
+    "@material/button": "^5.1.0",
+    "@material/checkbox": "^5.1.0",
+    "@material/chips": "^5.1.0",
+    "@material/data-table": "^5.1.0",
+    "@material/dialog": "^5.1.0",
+    "@material/drawer": "^5.1.0",
+    "@material/elevation": "^5.1.0",
+    "@material/fab": "^5.1.0",
+    "@material/floating-label": "^5.1.0",
+    "@material/form-field": "^5.1.0",
+    "@material/icon-button": "^5.1.0",
+    "@material/linear-ripple": "^5.1.0",
+    "@material/list": "^5.1.0",
+    "@material/radio": "^5.1.0",
+    "@material/select": "^5.1.0",
+    "@material/snackbar": "^5.1.0",
+    "@material/switch": "^5.1.0",
+    "@material/tab-scroller": "^5.1.0",
+    "@material/top-app-bar": "^5.1.0"
```
This is the wrong place for these dependencies. They should be moved to the package that's actually using these. If we're using RMWC, I'm not sure why we need to import these as dependencies at all. We can discuss this over a call if you're unsure what to do with these.

endymion1818 replied:
I moved these to `/packages/ui/package.json` and the build was successful. Sorry, my mistake I thought resolutions had to be at the top level `package.json` because the dependency `material-components-web` which also uses them is in this `package.json`.

Pavel910 commented on packages/app-admin/src/components/AppInstaller/styled.tsx:
```diff
@@ -39,11 +39,12 @@ export const InstallContent = styled("div")({
-        height: "100vh",
-        ".webiny-split-view__right-panel-wrapper": {
-            height: "100vh"
-        }
+        height: "100vh"
+    },
+    ".webiny-split-view__right-panel-wrapper": {
+        height: "100vh"
     }
 });
```
If I remember this correctly, this `100vh` should apply to both left and right panels. With this new rule it looks like you're only applying that height to the right panel.

endymion1818 replied:
This is just a syntax change, I've removed the nesting from the declaration as it wasn't getting evaluated.

Pavel910 commented on packages/app-form-builder/package.json:
```diff
@@ -19,7 +19,7 @@
     "@emotion/core": "^10.0.17",
     "@emotion/styled": "^10.0.17",
-    "@rmwc/menu": "^5.6.0",
+    "@rmwc/menu": "7.0.3",
```
This bothers me because `@rmwc` packages should only be used in the `ui` package. Please make a note of these packages, and we should solve this in a separate PR. All our apps should be using the `ui` package exclusively, to maintain the abstraction in one place.

endymion1818 replied:
Yep makes perfect sense

Pavel910 commented on packages/app-page-builder/package.json:
```diff
@@ -21,7 +21,7 @@
     "@fortawesome/free-solid-svg-icons": "^6.0.0",
     "@fortawesome/react-fontawesome": "^0.1.17",
-    "@rmwc/menu": "^5.6.0",
+    "@rmwc/menu": "7.0.3",
```
Same as the previous file.

endymion1818 replied:
👍🏻

Pavel910 commented on packages/cwp-template-aws/template/common/apps/theme/styles.scss:
```diff
@@ -1,3 +1,3 @@
 @import "./pageBuilder/styles/theme.scss";
 @import "./formBuilder/styles/theme.scss";
-@import "~@webiny/react-rich-text-renderer/styles.scss";
+@import "Z@webiny/react-rich-text-renderer/styles.scss";
```
A typo: `Z@webiny` :)

endymion1818 replied:
🙈

Pavel910 commented on packages/project-utils/bundling/app/config/webpack.config.js:
```diff
@@ -40,9 +40,11 @@ const cssRegex = /\.css$/;
 const sassLoader = {
     loader: require.resolve("sass-loader"),
     options: {
+        // webpackImporter: false,
```
Leave a comment why this is commented out.

endymion1818 replied:
It should be removed, I was doing a test to see if it materially changed anything but it didn't

Pavel910 commented on packages/ui/src/Chips/ChipIcon.tsx:
```diff
@@ -1,5 +1,5 @@
 import React from "react";
-import { ChipIcon as RmwcChipIcon, ChipIconProps as RmwcChipIconProps } from "@rmwc/chip";
+import { Chip as RmwcChipIcon, ChipProps as RmwcChipIconProps } from "@rmwc/chip";
```
Previously, there were 2 separate components: a `Chip`, and a `ChipIcon`. `ChipIcon` was used as a child of the `Chip` component to add an icon to a chip, here's an example:

```
<Chip>
    <ChipIcon icon={<AnIcon/>}/>
    Chip Label
</Chip>
```

This was changed ages ago, but this `<ChipIcon>` component was not removed. We don't use it anywhere in Webiny, so feel free to delete this component and stories for it. It's meaningless because the `<Chip>` itself supports `icon` props.

endymion1818 replied:
Done, thanks

Pavel910 reviewed (changes requested):
@endymion1818 did the first review so you can get some things sorted before jumping on a call. Let me know if any of my comments need clarification.

endymion1818 committed:
fix(ui): rm ChipIcon, responding to review comments

endymion1818 committed:
fix(ui): rm ChipIcon, responding to review comments

endymion1818 committed:
fix(ui): rm ChipIcon, responding to review comments

endymion1818 committed:
fix(ui): rm ChipIcon, responding to review comments

endymion1818 committed:
fix(app-admin): update select styles

endymion1818 committed:
fix(app-admin): update select styles

Pavel910 commented on packages/ui/package.json:
```diff
@@ -116,6 +116,30 @@
+    "@material/switch": "^5.1.0",
+    "@material/tab-scroller": "^5.1.0",
+    "@material/top-app-bar": "^5.1.0"
+  },
```
You cannot have `resolutions` per workspace. `resolutions` is only valid in the monorepo root, as their purpose is to resolve to one common dependency version for the entire project. And yarn tells you that when you try to install deps:

<img width="561" alt="CleanShot 2022-05-23 at 12 58 57@2x" src="https://user-images.githubusercontent.com/3920893/169805117-3cd37f47-59f4-4e01-85fb-b817940badf2.png">

My main question is: why do you need to specify these at all? Aren't these packages internal dependencies of @rmwc packages?

From what I see, @rmwc has these as explicit dependencies in their packages:
![CleanShot 2022-05-23 at 13 01 39@2x](https://user-images.githubusercontent.com/3920893/169805496-f648c0cd-8a76-4dbf-a074-6900be5cfc39.png)

This means that you don't need to specify any `@material/*` packages as Webiny dependencies at all.

endymion1818 replied:
Thanks for the convo about this, the issue is resolved by removing these dependencies. It turns out that updating the `material-components-web` package made it redundant.

endymion1818 committed:
fix(ui): add tslib as explicit dependency

endymion1818 committed:
fix(ui): add tslib as explicit dependency

Commit:
fix(package json): tidy resolutions, update lockfile

Commit:
fix(package json): tidy resolutions, update lockfile

endymion1818 committed:
fix(packagejson): pin tslib

endymion1818 committed:
fix(packagejson): pin tslib

brunozoric committed:
Merge branch 'next' into feat/apw-headless-cms

endymion1818 committed:
fix(packages): add tslib as explicit

endymion1818 committed:
fix(packages): add tslib as explicit

Pavel910 commented:
Found this weird behavior on Pages list filters:

https://user-images.githubusercontent.com/3920893/171034441-34a96d71-742e-4452-aab6-4bd7bbdea1bc.mp4

However, this works fine on Content Entries list filtering. Probably `select` options are not passed in the same way. RMWC 5 had some weird stuff going on with select boxes.

Pavel910 commented:
Another thing to check, something related to Portals:

<img width="1727" alt="CleanShot 2022-05-30 at 19 32 33@2x" src="https://user-images.githubusercontent.com/3920893/171039005-a53eafa5-dd8b-4e70-b3f5-1e319407dc41.png">

Add that `<Portal/>` component here: `packages/app-admin-rmwc/src/modules/Overlays/index.tsx:19`. There's a comment in that file describing what that stuff does.

Pavel910 commented:
Secondary button styles are slightly off, this is how they currently are, and with the new RMWC, the border is different.

![CleanShot 2022-05-30 at 22 57 22@2x](https://user-images.githubusercontent.com/3920893/171058573-1b36391b-27f0-4186-b280-85d934e15f65.png)

brunozoric committed:
Merge branch 'next' into feat/apw-headless-cms

endymion1818 committed:
fix(ui): restore secondary button styles

endymion1818 committed:
fix(ui): restore secondary button styles

endymion1818 committed:
feature(app-admin-rmwc): Add Portal to overlaysHOC

endymion1818 committed:
feature(app-admin-rmwc): Add Portal to overlaysHOC

endymion1818 committed:
fix(app-admin): update select input caret style

endymion1818 committed:
fix(app-admin): update select input caret style

endymion1818 committed:
fix(lockfile): sync lockfile

endymion1818 committed:
fix(lockfile): sync lockfile

endymion1818 committed:
chore(ui): prettify code

endymion1818 committed:
chore(ui): prettify code

endymion1818 committed:
fix(ui): rm tslib from main package.json

endymion1818 committed:
fix(ui): rm tslib from main package.json

endymion1818 committed:
chore(ui): move tslib to dependencies from devDependencies :face_palm:

endymion1818 committed:
chore(ui): move tslib to dependencies from devDependencies :face_palm:

endymion1818 committed:
fix(app-page-builder): add defaultValue & beforeChange to select

endymion1818 committed:
chore(packages): rm spurious file

endymion1818 committed:
chore(packages): rm spurious file

Pavel910 commented:
@endymion1818 Adding more screenshots to resolve:

The following three screenshots show the same issue, the floating label is positioned a bit off. Seems like it's only an issue with values prefilled by the browser. RMWC is probably not detecting that there's a value in the field (just guessing here).

![CleanShot 2022-06-05 at 16 01 35](https://user-images.githubusercontent.com/3920893/172054375-7f797430-a225-4c89-8878-7b93f9ed97ef.png)

![CleanShot 2022-06-05 at 16 02 34](https://user-images.githubusercontent.com/3920893/172054386-eda2cc27-e0df-4480-b5f4-351ae7177521.png)

![CleanShot 2022-06-05 at 16 04 28](https://user-images.githubusercontent.com/3920893/172054487-86db5c13-a0a1-44f3-a20a-c8162a75b38c.png)

Page Builder glitches:

![CleanShot 2022-06-05 at 16 05 39](https://user-images.githubusercontent.com/3920893/172054558-0e84a465-f49f-4df8-814e-e5842221e8e7.png)

Menu doesn't seem to work at all (that's probably that thing with Portal):

https://user-images.githubusercontent.com/3920893/172054602-132654e5-8cad-431a-91af-e9da1a4d4fdb.mp4

These buttons are off:

![CleanShot 2022-06-05 at 16 08 43](https://user-images.githubusercontent.com/3920893/172054661-21a46235-4f94-4950-ba0f-dd17131bff8f.png)

Pavel910 committed:
fix(ui): move tslib to admin app dependencies

Pavel910 committed:
fix(ui): move tslib to admin app dependencies

endymion1818 committed:
fix(ui): autofil label position

endymion1818 committed:
fix(ui): rm grid-gap from splitView

endymion1818 committed:
fix(app-page-builder): fix zoom select styles

endymion1818 committed:
fix(app-page-builder): icon positioning fix

endymion1818 committed:
fix(app-admin-rmwc): fix portal rendering issue

endymion1818 committed:
fix(app-admin-rmwc): fix portal rendering issue

endymion1818 committed:
fix(lockfile): rebase on next, resolve peer dependencies correctly

endymion1818 committed:
fix(lockfile): rebase on next, resolve peer dependencies correctly

endymion1818 committed:
fix(ui): pass data-testid as props to input component

endymion1818 committed:
fix(cypress): update installation script selectors

endymion1818 committed:
fix(ui): remove forked rmwc packages

endymion1818 committed:
fix(ui): remove forked rmwc packages

endymion1818 committed:
fix(app-admin-rwmc): synchronise dependencies

endymion1818 committed:
fix(app-admin-rwmc): synchronise dependencies

endymion1818 committed:
fix(cypress): update login tests, add testid to some views

endymion1818 committed:
fix(cypress): update selectors for uploadImage.spec

endymion1818 committed:
fix(cypress): update selectors and test

endymion1818 committed:
fix(cypress): update form builder tests

endymion1818 committed:
fix(cypress): add testids to fieldRenderers

endymion1818 committed:
fix(cyperss): update contentEntries tests

endymion1818 committed:
fix(cyperss): update contentEntries tests

endymion1818 committed:
fix(app-headless-cms): add testids to inputs

endymion1818 committed:
fix(cypress): update content models tests

endymion1818 committed:
fix(cypress): update content models tests

endymion1818 committed:
fix(cypress): update search and sort models

endymion1818 committed:
fix(cypress): update search and sort models

endymion1818 committed:
fix(cypress): i18n createLocale updates (wip)

endymion1818 committed:
fix(cypress): i18n createLocale updates (wip)

endymion1818 committed:
fix(cypress): update i18n tests

endymion1818 committed:
fix(cypress): update i18n tests

endymion1818 committed:
fix(cypress): update account tests

endymion1818 committed:
fix(cypress): update createMenu test

brunozoric committed:
Merge branch 'next' into feat/apw-headless-cms

# Conflicts:
#	packages/app-apw/package.json
#	yarn.lock

brunozoric committed:
fix(app-apw): form usages

adrians5j committed:
fix: bring back `AWS_ELASTIC_SEARCH_DOMAIN_NAME` env var support

adrians5j committed:
fix: revert `AWS_ELASTIC_SEARCH_DOMAIN_NAME` support

Pavel910 committed:
feat: add react-properties package (#2499)

Pavel910 committed:
fix(react-properties): use babel config from project-utils

adrians5j committed:
fix: add missing slash

adrians5j committed:
fix: add missing `advancedAccessControlLayer` property

adrians5j committed:
fix: ensure `WCP_PROJECT_ENVIRONMENT_API_KEY` env var

Pavel910 committed:
feat(pulumi): add support for remote resources

adrians5j committed:
Merge remote-tracking branch 'origin/next' into next

Pavel910 committed:
chore(pulumi): reformat with prettier

adrians5j committed:
fix: rename `PulumiRemoteResource` to `PulumiAppRemoteResource` [no ci]

adrians5j committed:
fix: add error handling

Commit:
chore(release): publish v5.29.0-beta.0 :tada: :rocket:

Commit:
chore: update package versions (webiny-versions)

Commit:
chore: update yarn.lock

Pavel910 committed:
fix(cypress): add a short wait to fix flaky test

Pavel910 committed:
fix(pulumi): process resource handlers at config stage

Pavel910 committed:
feat(pulumi-aws): add an AWS OpenSearch module

Pavel910 committed:
feat(pulumi-aws): add production features to ElasticSearch module

Pavel910 committed:
Merge branch 'v5' of github.com:webiny/webiny-js into v5

Commit:
chore(release): publish v5.29.0-beta.1 :tada: :rocket:

Commit:
chore: update package versions (webiny-versions)

Commit:
chore: update yarn.lock

Pavel910 committed:
fix(serverless-cms-aws): replace require with import [no ci]

adrians5j committed:
docs: replace `runProgram` with just `run` [no ci]

adrians5j committed:
chore: update `example.env`

adrians5j committed:
fix: convert `run` back to sync method

adrians5j committed:
chore: run prettier

Commit:
chore(release): publish v5.29.0-beta.2 :tada: :rocket:

Commit:
chore: update package versions (webiny-versions)

Commit:
chore: update yarn.lock

Commit:
chore(release): publish v5.29.0 :tada: :rocket:

Commit:
chore: update package versions (webiny-versions)

Commit:
chore: update yarn.lock

adrians5j committed:
ci: enable running Cypress tests on PRs

adrians5j committed:
ci: enable running Cypress tests on PRs

adrians5j committed:
ci: enable running Cypress tests on PRs

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

endymion1818 committed:
fix(cypress): pb menu items tests

endymion1818 committed:
fix(cypress): pb menu items tests

endymion1818 committed:
fix(cypress): update menus, users tests

endymion1818 committed:
fix(cypress): update menus, users tests

endymion1818 committed:
fix(cypress): update security admin group test

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
chore: update `jest` and `ts-jest` packages

adrians5j committed:
ci: enable running Cypress tests on PRs [no ci]

adrians5j committed:
ci: test Jest retries

adrians5j committed:
chore: add Jest setup file and globally set `jest.retryTimes(5);`

adrians5j committed:
ci: test Cypress retries

adrians5j committed:
ci: test Cypress retries

brunozoric committed:
Merge branch 'next' into feat/apw-headless-cms

# Conflicts:
#	apps/admin/src/App.tsx
#	apps/admin/src/plugins/apw.ts
#	apps/admin/src/plugins/index.ts
#	packages/app-apw/package.json
#	packages/app-apw/src/index.tsx
#	packages/app-apw/tsconfig.build.json
#	packages/app-apw/tsconfig.json
#	packages/project-utils/package.json
#	yarn.lock

adrians5j committed:
fix: update ts-jest

adrians5j committed:
fix: add `@jest-environment jsdom`

adrians5j committed:
chore: set `"jsx": "react"`

adrians5j committed:
chore: update yarn.lock

adrians5j committed:
Merge branch 'next' into ci-cd-pr-ctests

# Conflicts:
#	.github/workflows/pullRequestsCommandCypress.yml

adrians5j committed:
chore: run prettier

adrians5j committed:
chore: properly merge `setupFilesAfterEnv`

adrians5j committed:
chore: update babel-jest

adrians5j committed:
test: undo temporary change

adrians5j committed:
test: add `@jest-environment jsdom`

brunozoric committed:
refactor(api-headless-cms): optional chaning in checks (#2512)

endymion1818 committed:
fix(cypress): update pb menus list test

endymion1818 committed:
fix(cypress): update pb menus list test

endymion1818 committed:
fix(cypress): update pb categories tests

endymion1818 committed:
fix(lockfile): nuke lockfile and reinstall

endymion1818 committed:
fix(lockfile): nuke lockfile and reinstall

Pavel910 committed:
fix: unset default transform rules and remove obsolete name prop

Pavel910 committed:
chore: add a void package for types

Pavel910 committed:
fix: disable @types/jsdom package because of conflicts

Pavel910 committed:
test(api-apw): disable tenant link verification

Pavel910 committed:
test(react-properties): setup nanoid mocks

Pavel910 committed:
fix(cwp-template-aws): add missing .babelrc file

brunozoric committed:
feat(project-utils): use retries to delete indexes on snapshot errors (#2513)

adrians5j committed:
Merge branch 'next' into ci-cd-pr-ctests

adrians5j committed:
merge: pull request #2507

CI/CD - Enable Running Cypress Tests On PRs And Update Jest

adrians5j committed:
fix: use handlers for the `projectAppParams.pulumi` parameter

adrians5j committed:
fix: use handlers for the `projectAppParams.pulumi` parameter

adrians5j committed:
feat: export Pulumi app types

adrians5j committed:
Merge pull request #2516 from webiny/pulumi-handlers-fix

Use Handlers For `projectAppParams.pulumi` Parameter

adrians5j committed:
Merge pull request #2517 from webiny/pulumi-apps-types

Pulumi Apps - Exported Types

adrians5j committed:
chore: remove WCP project ID

brunozoric committed:
feat(api-apw): cms workflow (#2450)

adrians5j committed:
feat: assign WCP project ID via WCP_PROJECT_ID

Usually, the ID is set via webiny.project.ts, but, for dev/testing purposes, we want to be able to set the WCP project ID via an env var too.

brunozoric committed:
Merge branch 'next' into feat/apw-headless-cms

brunozoric committed:
feat: apw for headless cms

endymion1818 committed:
fix(cypress): rebase on next, resolve issues

endymion1818 committed:
fix(cypress): rebase on next, resolve issues

brunozoric committed:
feat(api-mailer): package to send e-mails from the system (#2514)

adrians5j committed:
fix: improve error messages

adrians5j committed:
feat: reorganize code and add support for WCP_PROJECT_ID env var

endymion1818 committed:
fix(lockfile): rebase on next, replace lockfile

endymion1818 committed:
fix(lockfile): rebase on next, replace lockfile

brunozoric committed:
feat(cwp-template-aws): add apw to user projects (#2523)

endymion1818 committed:
fix(packagejson): unpin pretty-format

endymion1818 committed:
fix(packagejson): unpin pretty-format

endymion1818 committed:
fix(app-page-builder): set default value for page zoom

adrians5j committed:
chore: run prettier

endymion1818 committed:
fix(ui): correct chip icon margins

brunozoric committed:
fix(app-headless-cms): revisions cache update on publish (#2525)

endymion1818 committed:
fix(app-admin): fix ui issue with SEO settings meta tags

endymion1818 committed:
fix(appformbuilder): update input label

brunozoric committed:
feat: typescript to 4.7.4 (#2527)

adrians5j commented:
/cypress

endymion1818 committed:
chore(package): rm tslib from package.json :ta_da:

endymion1818 committed:
chore(package): rm tslib from package.json :ta_da:

endymion1818 committed:
fix(cypress): update failed pipeline tests

endymion1818 commented:
/cypress

adrians5j committed:
ci: change permission-level to "write"

adrians5j committed:
Merge remote-tracking branch 'origin/next' into next

endymion1818 commented:
/cypress

endymion1818 committed:
fix(cypress): update downshift selectors

endymion1818 committed:
fix(cypress): update downshift selectors

brunozoric committed:
feat(api-headless-cms): combine cms contexts and graphqls (#2528)

leopuleo committed:
fix: menuItems Cypress tests

leopuleo committed:
fix: better child button handling

leopuleo committed:
fix: searchAndSortCategories Cypress tests

leopuleo committed:
chore: formatting

leopuleo committed:
fix: pageListMenuItemType Cypress tests

brunozoric committed:
refactor(api-apw): remove unnecessary cms code (#2530)

brunozoric committed:
test(api-headless-cms): header parameter plugin (#2532)

Pavel910 committed:
feat: decouple Page Editor and introduce a Block Editor (#2529)

endymion1818 committed:
chore(ui): rebase, format code

endymion1818 committed:
fix(cypress): fix remaining 2 tests

endymion1818 commented:
/cypress

endymion1818 commented:
/cypress

endymion1818 committed:
fix(cypress): improve flaky test

brunozoric committed:
feat(api-headless-cms): correct endpoint types (#2533)

endymion1818 committed:
fix(cypress): add pb page publish button testid

endymion1818 commented:
/cypress

Pavel910 committed:
fix(app-page-builder): resurrect "click-to-add" element (#2534)

leopuleo committed:
fix: cypress cross-origin tests

endymion1818 committed:
fix(cypress): select element correctly

leopuleo committed:
chore: code review and commands order

leopuleo committed:
chore: commands order

endymion1818 committed:
chore(cypres): wait for type to finish to reduce flakiness

endymion1818 commented:
/cypress

endymion1818 committed:
fix(cypress): force click confirm delete

endymion1818 commented:
/cypress

endymion1818 commented:
/cypress

brunozoric committed:
fix(api-headless-cms): locale error when system not installed (#2537)

endymion1818 commented:
/cypress

adrians5j committed:
feat: bring back APW into Webiny Projects (#2535)

* fix: bring back APW into Webiny projects

* chore: update yarn.lock

* fix: correct import of `AdvancedPublishingWorkflow`

* chore: regenerate TS configs

adrians5j committed:
fix(cli-watch-command): correct glob generation

* fix: remove empty if statement

* fix: correct glob generation

adrians5j committed:
fix: bring back APW into Webiny projects

* fix: bring back APW into Webiny projects

* chore: update yarn.lock

* fix: correct import of `AdvancedPublishingWorkflow`

* chore: regenerate TS configs

* fix: remove `AdvancedPublishingWorkflow` from template's `App.tsx` file

Pavel910 committed:
fix(app-page-builder): improve editor core (#2536)

Pavel910 commented on apps/admin/package.json:
```diff
@@ -28,7 +28,8 @@
     "regenerator-runtime": "^0.13.5",
-    "theme": "^0.1.0"
+    "theme": "^0.1.0",
+    "tslib": "^2.4.0"
```
@endymion1818 just to confirm: is `tslib` dependency still necessary? I know you mentioned that it's our rmwc forks that were causing issues, is that related?

endymion1818 replied:
@Pavel910 unfortunately yes, Im afraid we still get that issue with the `spreadArray` function we identified before without explicitly adding this line.

Pavel910 replied:
In that case, I assume we'll need to add this dependency as part of the project upgrade we're preparing for 5.30. Hopefully it's the only thing that needs changing in existing user projects. Have you tried upgrading from 5.29 ?

endymion1818 replied:
Ah, no haven't tested that. I'll run through it in the morning.

Pavel910 commented on packages/ui/src/Button/Button.styles.ts:
```diff
@@ -0,0 +1,16 @@
+ * Controls the button styles
+ * @type {string}
+ */
+const webinyButtonStyles = css(
+    {},
+    {
+        "&.webiny-ui-button--secondary:not(:disabled)": {
+            borderColor: "var(--mdc-theme-primary)"
+        }
+    }
+);
```
@endymion1818 This will look much nicer if you use the template literals; then you also don't need to use the camelCase for properties. You just write plain CSS instead.
```
const webinyButtonStyles = css`
    &.webiny-ui-button--secondary:not(:disabled) {
        border-color: var(--mdc-theme-primary)
    }
`
```

endymion1818 replied:
I was following what I thought was a convention but very happy you said this, I prefer template literals too.

Pavel910 commented on packages/ui/src/Input/Input.tsx:
```diff
@@ -37,8 +39,24 @@ export type InputProps = FormComponentProps &
+ * fix label position when autofilled
+ * @type {string}
+ */
+const webinyInputStyles = css(
+    {},
+    {
+        ".mdc-text-field__input:-webkit-autofill + .mdc-floating-label": {
+            transform: "translateY(-106%) scale(0.75)"
+        }
+    }
+);
```
Same as previous comment; try using template literals.

Pavel910 reviewed (changes requested):
@endymion1818 just a few minor things and we're good with the code review. The rest looks pretty solid. It's mostly adding of `data-testid` and some CSS tweaks across the board, so there isn't a whole lot of code changes that can break anything.

Pavel910 committed:
ci(cleanup): add EventBridge cleanup script

endymion1818 committed:
fix(ui): use template literals for css
